mitodl/open-discussions
# Display all topics in CompactCourseDisplay, make clickable

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #1888

#### What's this PR do?
- Displays all subject areas on the compact course card instead of just one.
- Makes each subject area on the card clickable which updates the search filters.

#### How should this be manually tested?
- Run a search
- On one of the cards, click a subject area.  That subject area's checkbox should get checked, the URL should get updated to include it, and the search should refresh with that filter applied.
- Uncheck the recently checked facet, it should be removed from the URL and the search results should refresh again without that filter applied.


#### Any background context you want to provide?
I ran into a strange problem where unchecking a facet checkbox that was recently activated via clicking on a CompactCourseCard would not trigger an `onChange` event.  This was resolved by altering the code to use `onClick` instead, though that does trigger a warning message in the console. 

#### Screenshots (if appropriate)
<img width="987" alt="screen shot 2019-03-04 at 12 33 37 pm" src="https://user-images.githubusercontent.com/187676/53751874-47eb1900-3e7b-11e9-98e8-321da2bcac25.png">

<img width="386" alt="screen shot 2019-03-04 at 12 33 58 pm" src="https://user-images.githubusercontent.com/187676/53751888-4c173680-3e7b-11e9-8783-4129af2ac160.png">
